File: projects/KD1OUR/automata/automata-scala/src/statemachine/scala/model/StateMachineState.scala
package statemachine.scala.model

/**
 * Az automata állapota (a szabály neve, avagy nemterminális szimbólumok)
 */
object StateMachineState extends Enumeration {

  type StateMachineState = Value

  val START_SIGNED,           // S
    START_UNSIGNED,           // S'
    ILLEGAL_STATE_0,          // I_0
    ILLEGAL_STATE_1,          // I_1 (e, vagy E karakter utáni állapot)
    LEGAL_STATE_1,            // T_1
    LEGAL_STATE_2,            // T_2
    LEGAL_STATE_3,            // T_3
    LEGAL_STATE_4,            // T_4
    LEGAL_STATE_5 = Value     // T_5

  def isFinalState(state: StateMachineState): Boolean ={
    state == LEGAL_STATE_1 ||
    state == LEGAL_STATE_2 ||
    state == LEGAL_STATE_3 ||
    state == LEGAL_STATE_4 ||
    state == LEGAL_STATE_5
  }

}

File: projects/KD1OUR/automata/automata-java/src/statemachine/java/model/StateMachineState.java
package statemachine.java.model;

/**
 * Az automata állapota (a szabály neve, avagy nemterminális szimbólumok)
 */
public enum StateMachineState {
    START_SIGNED(false),
    START_UNSIGNED(false),
    ILLEGAL_STATE_0(false),
    LEGAL_STATE_1(true),
    LEGAL_STATE_2(true),
    LEGAL_STATE_3(true),
    LEGAL_STATE_4(true),
    LEGAL_STATE_5(true),
    ILLEGAL_STATE_6(false),
    LEGAL_STATE_7(true);

    private boolean finalState;

    private StateMachineState(boolean finalState){
        this.finalState = finalState;
    }

    public boolean isFinalState(){
        return this.finalState;
    }
}
